src/tools/info/proto_info.c

```diff
@@ -99,34 +99,22 @@ static void listener_accept_callback(ucp_ep_h ep, void *arg)
     *(ucp_ep_h*)arg = ep;
 }
 
-static void set_saddr(const char *addr_str, uint16_t port, uint8_t ai_family,
-                      struct sockaddr *saddr)
+static void set_saddr(const char *addr_str, uint16_t port,
+                      sa_family_t af, struct sockaddr_storage *saddr)
 {
-    struct sockaddr_in *sa_in;
-    struct sockaddr_in6 *sa_in6;
-
-    memset(saddr, 0, sizeof(struct sockaddr_storage));
-    if (ai_family == AF_INET) {
-        sa_in = (struct sockaddr_in*)saddr;
-        if (addr_str != NULL) {
-            inet_pton(AF_INET, addr_str, &sa_in->sin_addr);
-        } else {
-            sa_in->sin_addr.s_addr = INADDR_ANY;
-        }
-        sa_in->sin_family = AF_INET;
-        sa_in->sin_port   = htons(port);
-    } else if (ai_family == AF_INET6) {
-        sa_in6 = (struct sockaddr_in6*)saddr;
-        if (addr_str != NULL) {
-            inet_pton(AF_INET6, addr_str, &sa_in6->sin6_addr);
-        } else {
-            sa_in6->sin6_addr = in6addr_any;
-        }
-        sa_in6->sin6_family = AF_INET6;
-        sa_in6->sin6_port   = htons(port);
+    memset(saddr, 0, sizeof(*saddr));
+
+    if (addr_str != NULL) {
+        /* coverity[check_return] */
+        ucs_sock_ipstr_to_sockaddr(addr_str, saddr);
+        ucs_assert(saddr->ss_family == af);
     } else {
-        fprintf(stderr, "Invalid address family");
+        saddr->ss_family = af;
+        /* coverity[check_return] */
+        ucs_sockaddr_is_inaddr_any((struct sockaddr*)saddr);
```

Comment:
what is this doing? seems no effect

Reply:
setting INADDR_ANY/in6addr_any incase IP is not specified (listener).  it passed hardcoded "0.0.0.0" earlier.

Reply:
needed for set INADDR_ANY/in6addr_any incase IP is is not specified (listener). it was passing hardcoded "0.0.0.0" earlier.